src/gd_tga.c

```diff
@@ -237,7 +237,10 @@ int read_image_tga( gdIOCtx *ctx, oTga *tga )
 			return -1;
 		}
 
-		gdGetBuf(conversion_buffer, image_block_size, ctx);
+		if (gdGetBuf(conversion_buffer, image_block_size, ctx) != image_block_size) {
+			gdFree(conversion_buffer);
```

Comment:
Maybe we should add an error message here (and also for the RLE code below):

    gd_error("gd-tga: premature end of image data\n");